src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BasePatternSwitchLocalRewriter.cs

```diff
@@ -0,0 +1,415 @@
+                }
+
+                foreach (BoundDecisionDag node in sortedNodes)
+                {
+                    switch (node)
+                    {
+                        case BoundWhenClause w:
+                            GetDagNodeLabel(node);
+                            //Debug.Assert(w.WhenTrue != null);
+                            //GetDagNodeLabel(w.WhenTrue);
+                            if (w.WhenFalse != null)
+                            {
+                                GetDagNodeLabel(w.WhenFalse);
+                                _backwardLabels.Add(w.WhenFalse);
+                            }
+                            break;
+                        case BoundDecision d:
+                            // Final decisions can branch directly to the target
+                            _dagNodeLabels[node] = d.Label;
+                            break;
+                        case BoundEvaluationPoint e:
+                            notePredecessor(e.Next);
+                            break;
+                        case BoundDecisionPoint p:
+                            notePredecessor(p.WhenTrue);
+                            notePredecessor(p.WhenFalse);
+                            break;
+                        default:
+                            throw ExceptionUtilities.UnexpectedValue(node.Kind);
+                    }
+                }
+
+                hasPredecessor.Free();
+            }
+
+            protected new void Free()
+            {
+                base.Free();
+                _loweredDecisionDag.Free();
+                _backwardLabels.Free();
+            }
+
+            protected virtual LabelSymbol GetDagNodeLabel(BoundDecisionDag dag)
+            {
+                if (!_dagNodeLabels.TryGetValue(dag, out LabelSymbol label))
+                {
+                    _dagNodeLabels.Add(dag, label = dag is BoundDecision d ? d.Label : _factory.GenerateLabel("dagNode"));
+                }
+
+                return label;
+            }
+
+            /// <summary>
+            /// Lower the given nodes into _loweredDecisionDag.
+            /// </summary>
+            private void LowerDecisionDag(ImmutableArray<BoundDecisionDag> sortedNodes)
+            {
+                switch (sortedNodes[0])
+                {
+                    case BoundWhenClause wc:
+                        // If the first node is in a when clause's section rather than the code for the
+                        // lowered decision dag, jump there to start.
+                        _loweredDecisionDag.Add(_factory.Goto(GetDagNodeLabel(wc)));
+                        break;
+                    case BoundDecision d:
+                        // If the first node is a decision rather than the code for the
+                        // lowered decision dag, jump there to start.
+                        _loweredDecisionDag.Add(_factory.Goto(GetDagNodeLabel(d)));
+                        break;
+                }
```

Comment:
>} [](start = 16, length = 1)

Consider merging the two cases.

Reply:
Recorded in [https://github.com/dotnet/roslyn/issues/25663](https://github.com/dotnet/roslyn/issues/25663)

---
In reply to: [176304700](https://github.com/dotnet/roslyn/pull/25544#discussion_r176304700) [](ancestors = 176304700)